roles/repositories/tasks/main.yml

```diff
@@ -43,7 +43,7 @@
     gpg_key: "{{ item.gpg_key | default(omit) }}"
     state: present
   with_items:
-    - "{{ foreman_products | selectattr('repositories', 'defined') | list }}"
+    - "{{ foreman_products | selectattr('repositories', 'defined') | map('combine', {'repositories': '[FILTERED]'})| list }}"
```

Comment:
the alternative is to unconditionally set `no_log: true`, as "finding out whether any of the defined repos has a password set" is way to complicated in this step (it's easy in the next one, where we actually iterate over the repos).